lib/bitex_bot/models/api_wrappers/bitfinex/bitfinex_api_wrapper.rb

```diff
@@ -1,119 +1,129 @@
+##
+# Wrapper implementation for Bitfinex API.
+# https://docs.bitfinex.com/docs
+#
 class BitfinexApiWrapper < ApiWrapper
-  cattr_accessor :max_retries do 1000 end
-
-  def self.setup(settings)
-    Bitfinex::Client.configure do |conf|
-      conf.api_key = settings.bitfinex.api_key
-      conf.secret = settings.bitfinex.api_secret
+  cattr_accessor(:max_retries) { 1000 }
+
+  class << self
+    def setup(settings)
+      Bitfinex::Client.configure do |conf|
+        conf.api_key = settings.bitfinex.api_key
+        conf.secret = settings.bitfinex.api_secret
+      end
     end
-  end
 
-  def self.transactions
-    with_retry 'transactions' do
-      client.trades.map { |t| transaction_parser(t.symbolize_keys) }
+    def transactions
+      with_retry :transactions do
+        client.trades.map { |t| transaction_parser(t.symbolize_keys) }
+      end
     end
-  end
 
-  def self.orders
-    with_retry 'orders' do
-      client.orders.map { |o| order_parser(o.symbolize_keys) }
+    def orders
+      with_retry :orders do
+        client.orders.map { |o| order_parser(o.symbolize_keys) }
+      end
     end
-  end
 
-  def self.order_book
-    with_retry 'order_book' do
-      book = client.orderbook.deep_symbolize_keys
-      order_book_parser(book)
+    def order_book
+      with_retry :order_book do
+        order_book_parser(client.orderbook.deep_symbolize_keys)
+      end
     end
-  end
 
-  def self.balance
-    with_retry 'balance' do
-      balances = client.balances(type: 'exchange').map(&:symbolize_keys)
-      balance_summary_parser(balances)
+    def balance
+      with_retry :balance do
+        balances = client.balances(type: 'exchange').map(&:symbolize_keys)
+        balance_summary_parser(balances)
+      end
     end
-  end
 
-  # We don't need to fetch the list of transactions for bitfinex
-  def self.user_transactions
-    []
-  end
-
-  def self.place_order(type, price, quantity)
-    with_retry "place order #{type} #{price} #{quantity}" do
-      order_data =
-        Bitfinex::Client.new
-        .new_order('btcusd', quantity.round(4), 'exchange limit', type.to_s, price.round(2))
-      BitfinexOrder.new(order_data)
+    # We don't need to fetch the list of transactions for bitfinex
+    def user_transactions
+      []
     end
-  end
 
-  def self.amount_and_quantity(order_id, transactions)
-    with_retry "find order #{order_id}" do
-      order = Bitfinex::Client.new.order_status(order_id)
-      [order['avg_execution_price'].to_d * order['executed_amount'].to_d, order['executed_amount'].to_d]
+    def place_order(type, price, quantity)
+      with_retry "place order #{type} #{price} #{quantity}" do
+        order_data =
+          Bitfinex::Client.new.new_order('btcusd', quantity.round(4), 'exchange limit', type.to_s, price.round(2))
```

Comment:
Any particular reason why we're instantiating a new client instead of using `self.client`?

Reply:
typooo